packages/core/src/render3/errors.ts

```diff
@@ -70,6 +75,21 @@ export function throwErrorIfNoChangesMode(
   throw new RuntimeError(RuntimeErrorCode.EXPRESSION_CHANGED_AFTER_CHECKED, msg);
 }
 
+function formatValue(value: any): string {
+  let strValue: string = String(value);
+
+  // JSON.stringify will throw on circular references
+  try {
+    if (strValue === '[object Object]') {
```

Comment:
I think we can also use `JSON.stringify` for arrays:
```suggestion
    if (Array.isArray(value) || strValue === '[object Object]') {
```